src/observers/strategies/cache_occupancy.cpp

```diff
@@ -0,0 +1,81 @@
+#include <list>
+#include <utility>
+#include <vector>
+
+#include "bus/event_bus.hpp"
+
+#include "observers/strategies/seniority.hpp"
+
+#include "serenity/resource_helper.hpp"
+
+#include "cache_occupancy.hpp"
+
+namespace mesos {
+namespace serenity {
+
+using std::list;
+using std::pair;
+
+Try<QoSCorrections> CacheOccupancyStrategy::decide(
+    ExecutorAgeFilter* ageFilter,
+    const Contentions& contentions,
+    const ResourceUsage& usage) {
+  std::vector<ResourceUsage_Executor> cmtEnabledExecutors =
```

Comment:
Vector is continuous block in memory. Continuous block have better chance to by wholly slurped into cache line in one memory read.